// Copyright 2017 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package clif_example.wrap_protos.protos;


message MyMessage {
  optional string name = 1;
  optional EmbeddedMsg msg = 2;

  message Nested {
    enum NestedEnum {
      DEFAULT = 0;
      EXPLICIT = 1;
    };
    optional NestedEnum value = 1;
  };

  optional Nested nested = 3;
};

message EmbeddedMsg {
  repeated uint32 id = 1;
};
